ott/core/potentials.py

```diff
@@ -49,29 +61,32 @@ def transport(self, vec: jnp.ndarray, forward: bool = True) -> jnp.ndarray:
       The transported points.
     """
     vec = jnp.atleast_2d(vec)
-    if self._cor:
+    if self._corr and isinstance(self.cost_fn, costs.SqEuclidean):
       return self._grad_g(vec) if forward else self._grad_f(vec)
-    return vec - 0.5 * (self._grad_f(vec) if forward else self._grad_g(vec))
+    grad_h_inv = jax.vmap(jax.grad(self.cost_fn.h_legendre))
```

Comment:
I'd refactor this as
```python
@property
def _grad_h_inv(self) -> Callable[[jnp.ndarray], jnp.ndarray]:
    assert isinstance(self.cost_fn, costs.RBFCost), "Some nice message."
    return jax.vmap(jax.grad(self.cost_fn.h_legendre))
```

Reply:
ok! this will still not return an error is `h_legendre` is not implemented.